docs/docs/Indexing_JSON.md

```diff
@@ -1,216 +1,367 @@
 
-Given the following document:
+Use the optional `TAGS` keyword to specify the strings that will surround (or highlight) the matching search terms.
 
-```JSON
-{
-  "name": "Headquarters",
-  "address": [
-    "Suite 250",
-    "Mountain View"
-  ],
-  "cp": "CA 94040"
-}
+For example, highlight the word "bluetooth" with bold HTML tags in item names and descriptions:
+
+```sql
+127.0.0.1:6379> FT.SEARCH itemIdx '(@name:(bluetooth))|(@description:(bluetooth))' RETURN 3 name description price HIGHLIGHT FIELDS 2 name description TAGS '<b>' '</b>'
+1) "2"
+2) "item:1"
+3) 1) "name"
+   2) "Noise-cancelling <b>Bluetooth</b> headphones"
+   3) "description"
+   4) "Wireless <b>Bluetooth</b> headphones with noise-cancelling technology"
+   5) "price"
+   6) "99.98"
+4) "item:2"
+5) 1) "name"
+   2) "Wireless earbuds"
+   3) "description"
+   4) "Wireless <b>Bluetooth</b> in-ear headphones"
+   5) "price"
+   6) "64.99"
 ```
 
-Before you can index the array under the `address` key, you have to create two fields:
+## Aggregate with JSONPath
 
-```SQL
-FT.CREATE orgIdx ON JSON SCHEMA $.address[0] AS a1 TEXT $.address[1] AS a2 TEXT
-OK
-```
+You can use [aggregation](/docs/stack/search/reference/aggregations) to generate statistics or build facet queries.
 
-You can now index the document:
+The `LOAD` option accepts [JSONPath](/docs/stack/json/path) expressions. You can use any value in the pipeline, even if the value is not indexed.
 
-```SQL
-JSON.SET org:1 $ '{"name": "Headquarters","address": ["Suite 250","Mountain View"],"cp": "CA 94040"}'
-OK
+This example uses aggregation to calculate a 10% price discount for each item and sorts the items from least expensive to most expensive:
+
+```sql
+127.0.0.1:6379> FT.AGGREGATE itemIdx '*' LOAD 4 name $.price AS originalPrice APPLY '@originalPrice - (@originalPrice * 0.10)' AS salePrice SORTBY 2 @salePrice ASC
+1) "2"
+2) 1) "name"
+   2) "Wireless earbuds"
+   3) "originalPrice"
+   4) "64.99"
+   5) "salePrice"
+   6) "58.491"
+3) 1) "name"
+   2) "Noise-cancelling Bluetooth headphones"
+   3) "originalPrice"
+   4) "99.98"
+   5) "salePrice"
+   6) "89.982"
 ```
 
-You can now search in the address:
+{{% alert title="Note" color="info" %}}
+`FT.AGGREGATE` queries require `attribute` modifiers. Don't use JSONPath expressions in queries, except with the `LOAD` option, because the query parser doesn't fully support them.
+{{% /alert %}}
 
-```SQL
-FT.SEARCH orgIdx "suite 250"
-1) (integer) 1
-2) "org:1"
-3) 1) "$"
-   2) "{\"name\":\"Headquarters\",\"address\":[\"Suite 250\",\"Mountain View\"],\"cp\":\"CA 94040\"}"
-```
+## Index limitations
+
+### Schema mapping
+
+During index creation, you need to map the JSON elements to `SCHEMA` fields as follows:
 
-### Index JSON strings and numbers as TEXT and NUMERIC
+- Strings as `TEXT`, `TAG`, or `GEO`.
+- Numbers as `NUMERIC`.
+- Booleans as `TAG`.
+- JSON array of strings or booleans in a `TAG` field. Other types (`NUMERIC`, `GEO`, `NULL`) are not supported.
```

Comment:
@rrelledge JSON array of boolean values is not yet supported (PR #2940)

Reply:
@oshadmi I will update this in the new PR as well.

Reply:
This update is include in this PR: https://github.com/RediSearch/RediSearch/pull/2949